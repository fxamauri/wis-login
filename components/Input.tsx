import React from 'react';

import * as S from './Input.styles';

type Props = {
  name: string;
  label?: string;
  isInvalid?: boolean;
  invalidMessage?: string;
};

const Input = ({
  name,
  label,
  isInvalid,
  invalidMessage = '',
  ...props
}: Props & React.InputHTMLAttributes<HTMLInputElement>) => {
  return (
    <S.Container>
      {label && <S.Label htmlFor={name}>{label}</S.Label>}
      <S.InputContainer isInvalid={isInvalid}>
        <S.InputStyled data-testid="input" name={name} {...props} />
        {isInvalid && <S.CloseIcon />}
      </S.InputContainer>
      {isInvalid && <S.ErrorMessage data-testid="input-message">{invalidMessage}</S.ErrorMessage>}
    </S.Container>
  );
};

export default Input;
