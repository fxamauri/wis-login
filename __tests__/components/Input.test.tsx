import { render } from '@testing-library/react';
import Input from '../../components/Input';
describe('Input', () => {
  it('Should render Input', () => {
    const { container } = render(<Input name="password" label="Password" isInvalid={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render with invalid message', () => {
    const { getByTestId } = render(
      <Input
        name="password"
        label="Password"
        isInvalid={true}
        invalidMessage="there was an error"
      />,
    );

    const inputMessage = getByTestId('input-message');
    expect(inputMessage).toBeInTheDocument();
    expect(inputMessage.textContent).toBe('there was an error');
  });
});
