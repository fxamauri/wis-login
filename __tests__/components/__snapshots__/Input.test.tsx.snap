// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render Input 1`] = `
<div
  class="Inputstyles__Container-sc-17xchd0-0 bOXAkU"
>
  <label
    class="Inputstyles__Label-sc-17xchd0-1 iYXcqo"
    for="password"
  >
    Password
  </label>
  <div
    class="Inputstyles__InputContainer-sc-17xchd0-3 iUQGbd"
  >
    <input
      class="Inputstyles__InputStyled-sc-17xchd0-2 ieXVSx"
      data-testid="input"
      name="password"
    />
  </div>
</div>
`;
